glean-core/src/lib.rs

```diff
@@ -183,12 +184,13 @@ impl Glean {
             event_data_store,
             core_metrics: CoreMetrics::new(),
             internal_pings: InternalPings::new(),
-            data_path: PathBuf::from(cfg.data_path),
+            data_path: PathBuf::from(&cfg.data_path),
             application_id,
             ping_registry: HashMap::new(),
             start_time: local_now_with_offset(),
             max_events: cfg.max_events.unwrap_or(DEFAULT_MAX_EVENTS),
             is_first_run: false,
+            upload_manager: PingUploadManager::new(&cfg.data_path),
```

Comment:
```suggestion
            #[cfg(feature = "upload")]
            upload_manager: PingUploadManager::new(&cfg.data_path),
```

Would this work?

Reply:
It would work, the problem is I'd have to spread these around all over the code.

Reply:
Not sure this is a problem. But maybe @badboy has another way to still do this while not enabling this on master

Reply:
I did it ec7aae6 :)